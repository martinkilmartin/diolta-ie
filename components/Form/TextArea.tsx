type Props = {
  title?: string
  name?: string
  id?: string
  placeholder: string
}

const TextArea = ({ title, name, placeholder, id }: Props): JSX.Element => (
  <div className="form-control">
    {title && (
      <label className="label">
        <span>{title}</span>
      </label>
    )}
    <textarea
      className="h-24 textarea textarea-bordered textarea-primary"
      placeholder={placeholder}
      name={name}
      id={id}
    ></textarea>
  </div>
)

export default TextArea
