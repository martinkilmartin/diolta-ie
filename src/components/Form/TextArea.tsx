type Props = {
  title?: string
  name?: string
  id?: string
  minLength?: number
  maxLength?: number
  placeholder: string
}

const TextArea = ({
  title,
  name,
  minLength = 3,
  maxLength = 10000,
  placeholder,
  id,
}: Props): JSX.Element => (
  <div className="form-control">
    {title && (
      <label className="label">
        <span>{title}</span>
      </label>
    )}
    <textarea
      className="h-24 textarea textarea-bordered textarea-primary"
      placeholder={placeholder}
      name={name}
      id={id}
      minLength={minLength}
      maxLength={maxLength}
      required
    ></textarea>
  </div>
)

export default TextArea
