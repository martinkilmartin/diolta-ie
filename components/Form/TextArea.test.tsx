import renderer from 'react-test-renderer'
import TextArea from './TextArea'

test('TextArea displays label and text-area-input', () => {
  const title = 'Message'
  const placeholder = 'Enter your message here'
  const textAreaRenderer = renderer.create(
    <TextArea title={title} placeholder={placeholder} />
  )
  const tree = textAreaRenderer.toJSON()
  expect(tree).toMatchSnapshot()
  const textAreaInstance = textAreaRenderer.root
  expect(
    textAreaInstance.findByProps({
      className: 'label-text',
    }).children[0]
  ).toEqual(title)
  expect(
    textAreaInstance.findByProps({
      className: 'h-24 textarea textarea-bordered textarea-primary',
    }).props.placeholder
  ).toEqual(placeholder)
})

test('TextArea displays text-area-input without label', () => {
  const placeholder = 'Enter your message here'
  const textAreaRenderer = renderer.create(
    <TextArea placeholder={placeholder} />
  )
  const tree = textAreaRenderer.toJSON()
  expect(tree).toMatchSnapshot()
  const textAreaInstance = textAreaRenderer.root
  expect(
    textAreaInstance.findByProps({
      className: 'form-control',
    }).children.length
  ).toEqual(1)
  expect(
    textAreaInstance.findByProps({
      className: 'h-24 textarea textarea-bordered textarea-primary',
    }).props.placeholder
  ).toEqual(placeholder)
})
