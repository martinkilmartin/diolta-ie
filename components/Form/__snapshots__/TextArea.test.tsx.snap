// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea displays label and text-area-input 1`] = `
<div
  className="form-control"
>
  <label
    className="label"
  >
    <span
      className="label-text"
    >
      Message
    </span>
  </label>
  <textarea
    className="h-24 textarea textarea-bordered textarea-primary"
    placeholder="Enter your message here"
  />
</div>
`;

exports[`TextArea displays text-area-input without label 1`] = `
<div
  className="form-control"
>
  <textarea
    className="h-24 textarea textarea-bordered textarea-primary"
    placeholder="Enter your message here"
  />
</div>
`;
